test/e2e-common.sh

```diff
@@ -94,13 +95,15 @@ KAFKA_CLUSTER_URL=${KAFKA_PLAIN_CLUSTER_URL}
 # Kafka channel CRD config template file. It needs to be modified to be the real config file.
 readonly KAFKA_CRD_CONFIG_TEMPLATE="400-kafka-config.yaml"
 
-# Real Kafka channel CRD config, generated from the template directory and modified template file.
-readonly KAFKA_CRD_CONFIG_DIR="$(mktemp -d)"
-# Real Kafka Source CRD config, generated from the template directory and modified template file.
-readonly KAFKA_SOURCE_CRD_CONFIG_DIR="$(mktemp -d)"
-
-# Remove the temporary directories on exit (avoiding "rm -rf" to prevent disaster if something is wrong with the variables)
-trap "{ for dirrm in \"${KAFKA_CRD_CONFIG_DIR}\" \"${KAFKA_SOURCE_CRD_CONFIG_DIR}\"; do rm \"\${dirrm}\"/*; rmdir \"\${dirrm}\"; done }" EXIT
+# Real Kafka channel CRD config, generated from the template directory and
+# modified template file.
+readonly KAFKA_CRD_CONFIG_DIR="$(mktemp -d "${ARTIFACTS}/channel-crd-XXXXX")"
+# Real Kafka Source CRD config, generated from the template directory and
+# modified template file.
+readonly KAFKA_SOURCE_CRD_CONFIG_DIR="$(mktemp -d "${ARTIFACTS}/source-crd-XXXXX")"
+# A target directory where post install scripts are gets created.
+readonly KAFKA_POST_INSTALL_DIR="$(mktemp -d "${ARTIFACTS}/post-install-XXXXX")"
```

Comment:
Not sure I get the `ARTIFACTS` here ?

Reply:
`ARTIFACTS` variable is already defined for all knative projects. So creating those temp files in there is beneficial as you can look up them afterwards on Prow. 

Example, here: https://console.cloud.google.com/storage/browser/knative-prow/pr-logs/pull/knative-sandbox_eventing-kafka/467/pull-knative-sandbox-eventing-kafka-integration-test-channel-consolidated/1377556813607604224/artifacts/7d49bde6-92ce-11eb-8245-72fd3a7afd21/channel-crd-3xPze?pageState=(%22StorageObjectListTable%22:(%22f%22:%22%255B%255D%22))&prefix=&forceOnObjectsSortingFiltering=false